<launch>
    <arg name="bottom_laser" value="0.4859 0 0.1248 3.142 0 3.142 base_link bottom_laser_link 10"/>
    <arg name="bottom_lidar_filter_enable"    value="$(optenv BOTTOM_LIDAR_FILTER_ENABLE false)"/>

    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find rplidar_ros)/configs/custom_rosconsole.conf"/>

    <node pkg="tf" type="static_transform_publisher" name="bottom_laser_tf" args="$(arg bottom_laser)" />

    <node name="bottom_rplidar_node" pkg="rplidar_ros" type="rplidarNode" output="screen">
        <param name="channel_type"        type="string" value="udp"/>  
        <param name="udp_ip"              type="string" value="10.7.5.151"/>  
        <param name="frame_id"            type="string" value="bottom_laser_link"/>
        <param name="inverted"            type="bool"   value="false"/>
        <param name="angle_compensate"    type="bool"   value="true"/>
        <param name="scan_mode"           type="string" value="Sensitivity"/>
        <param name="scan_frequency"      type="int"  value="10"/>
        <remap from="scan" to="bottom_laser_scan" />
    </node>
    
    <group if="$(arg bottom_lidar_filter_enable)">
        <node pkg="laser_filters" type="scan_to_scan_filter_chain" name="laser_filter_right">
            <rosparam command="load" file="$(find rplidar_ros)/configs/bottom_lidar_filter_right.yaml" />
            <remap from="scan" to="bottom_laser_scan" />
            <remap from="scan_filtered" to="bottom_laser_scan_right" />
        </node>

        <node pkg="laser_filters" type="scan_to_scan_filter_chain" name="laser_filter_left">
            <rosparam command="load" file="$(find rplidar_ros)/configs/bottom_lidar_filter_left.yaml" />
            <remap from="scan" to="bottom_laser_scan" />
            <remap from="scan_filtered" to="bottom_laser_scan_left" />
        </node>

        <node name="laserscan_multi_merger" pkg="ira_laser_tools" type="laserscan_multi_merger" output="screen" respawn="true">
            <param name="destination_frame" value="bottom_laser_link"/>
            <param name="scan_destination_topic" value="/bottom_laser_scan_filtered"/>
            <param name="laserscan_topics" value="/bottom_laser_scan_left /bottom_laser_scan_right"/>
            <param name="angle_min" type="double" value="-3.142"/>
            <param name="angle_max" type="double" value="3.142"/>
            <param name="range_min" type="double" value="0.15"/>
            <param name="range_max" type="double" value="30.0"/>
        </node>
    </group>
</launch>
  
